cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -870,11 +916,15 @@ void reader::impl::allocate_nesting_info()
             }
           }
 
+          // Get the schema from the current input source.
+          auto& actual_cur_schema = _metadata->get_schema(
```

Comment:
Locally get the actual current schema in use instead of messing with the `cur_schema`